### ISSUE TITLE
Gallery wiped majority of favorited media

### ISSUE THREAD

**Issue by mgrandi** [FossifyOrg/Gallery#383]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Gallery/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Gallery/issues) or [closed](https://github.com/FossifyOrg/Gallery/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Gallery/discussions) or [closed](https://github.com/FossifyOrg/Gallery/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Gallery/blob/master/CONTRIBUTING.md).

### Affected app version

1.2.1

### Affected Android/Custom ROM version

Android 15

### Affected device model

pixel 7 pro 

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

Was viewing favorites 
Moved away from the app
Came back to the app
Favorites media went from thousands to 30 entries 

### Expected behavior

To not wipe my favorites 

### Actual behavior

It wiped all but 30 of my favorited media

### Screenshots/Screen recordings

No recordings

### Additional information

This is an existing issue I found with SimpleMediaTools, where for some reason some how, the favorites storage is corrupted and it handles this by just nuking the entire or most of the favorites list, which is incredibly annoying and makes me not want to use this feature at all, as it has happened at least twice with simple media tools, and now once with the new version

**Comment by Aga-C** [comment]
It's possible to revert all your favorites by entering again into folders from where you've added media to favorites.

Also, here are some tips for easy bug reproduction:
- It seems that it appears when there are many directories with many files. I was only able to reproduce it on a phone with 32 folders with circa 2500 files.
- It's easiest to reproduce by adding files from multiple directories. I couldn't reproduce it by adding files only from one or two directories.

**Comment by mgrandi** [comment]
I force quit the app a few times but after a few times and navigating to the folder again it seems to have come back

This smells like a race condition with it not saving the favorites list in a safe manner, because a few times when this happened with simple media tools gallery, the favorites list did not come back at all

**Comment by naveensingh** [comment]
Couldn't reproduce this with thousands of files and 30+ folders. Does anybody have more tips?

**Comment by albert-softie** [comment]
I started running into this bug again recently after I had moved some favorited images to the recycle bin, and out of curiosity I tried restoring them and the favorites gallery was suddenly fixed and no longer wiping itself when refreshing in various folders. Since I've has the bug happen before when not using the recycle bin, I'm guessing it has something to do with deleting favorited images in general

**Comment by Lppsoeht** [comment]
I can confirm this(for me) seems to happen only when some fav files get moved/deleted externally, not by the app. After moving once I open the app it breaks favs folder.

The thing if this can't get fixed would be caching a backup file and allow the user to restore from the last known "working" favourites list. But idk since I also tried to import a old favs backup list but it seems to do nothing.

My fav list today went from hundreds to 12, the substraction isn't just about the moved/deleted files.

Returning to some folders forced the favs refresh and I assume cycling through every folder will restore all.

**Comment by Lppsoeht** [comment]
There are surely some bugs regarding folders refresh/caching when there are a lot of media files.
For example, now the favorites folders doesn't seem to remember the big folder I forced refresh minutes ago, from 300 favs from that refresh I then opened a smaller folder with like 5 favs and the favs folder got from 312 to 17

**Comment by alice-fish** [comment]
It seems when exporting the favourite list, it only includes images which properly appear in Favourites. This becomes an issue when I have multiple large folders, many of which having favourited media often not appearing in Favourites, thus favourite list cannot be exported properly in one go.

**Comment by EhudKirsh** [comment]
I have this issue as well. I encountered it as recently as yesterday, and also several times before over the last few months.
Please consider making some quick fixes for it, like a button that loads all folders in the background, or at least the paths in the `.txt` backup file of Fossify Gallery's favorites. Just to be able to load up all the favorites again, to be able to add or remove more images, and export.
Basically, have a hot fix that returns the normal functionality. Right now it's not usable.
I won't manually enter dozens of folders just to load up all the favorites and risk missing a folder as I make changes and export.

**Comment by HansCronau** [comment]
I have experienced this in version 1.13.1 as well.

@Aga-C's theory about [many favourites in many folders](https://github.com/FossifyOrg/Gallery/issues/383#issuecomment-2563996838) seems plausible in my case as well:
My favourites used to consist of mostly pictures in `Camera/DCIM` and `Pictures`. Last month I added a new folder with a quickly growing number of favourites (easily over a hundred). Before this I never had any issues, although ai cannot rule out coincidence.
(I would have loved to double check if I had favourites in other folders as well —probably some screenshots too— but I cannot, for obvious reasons.)

I cannot 100% confirm @Lppsoeht's theory about it [being triggered by external favourite deletion](https://github.com/FossifyOrg/Gallery/issues/383#issuecomment-4231619640), but I too was deleting files externally. I did not _intend_ to delete any favourites, but I may very well have by accindent.

What may be Interesting is the way in which my Favorites view broke:

* I had been deleting some files in `Camera/DCIM`.
* I went to the Gallery app and opened Favorites.
* Initially I saw all my Favorites, but they disappeared while the screen was open and only favourites in `Camera/DCIM` remained.

What may also be interesting is what has happened since:

* Unfortunately browsing or refreshing (pull down) other folders (than `Camera/DCIM`) with Gallery has done nothing to repair lost favourites in those folders.
* I initially did not see _any_ items as starred in the folder view of other folders than `Camera/DCIM`, _even if I did favourite them_. This _very much_ looks to be a caching issue, because after giving it some time, some newly favourited items did become starred in the folder view, yet they _remained_ starred in the folder view after again removing them from favourites (via the single item view). This too could be fixed by either giving it some time (half a minute), or by actively refreshing (pull down) the folder. I do not remember this being as slow in the past.
* Even items newly favourited in `Camera/DCIM` at some point _suddenly_ disappeared from the Favorites view, only to reappear a few minutes later. Note that this happened exclusively to items in `Camera/DCIM` that were favourited _after_ issues started. I do not know what this hints towards, but I did find it peculiar.

Hope this is of any help.
